lib/Doctrine/DBAL/Driver/PDOStatement.php

```diff
@@ -88,7 +98,13 @@ public function bindParam($column, &$variable, $type = \PDO::PARAM_STR, $length
     public function execute($params = null)
     {
         try {
-            return parent::execute($params);
+            $result = parent::execute($params);
+
+            if (null !== $this->connection) {
```

Comment:
:+1: BTW I am still not using either style consistently. Do you have any reason why to prefer one over the other?